crates/miden-lib/src/account/interface/mod.rs

```diff
@@ -176,6 +135,122 @@ impl From<&Account> for AccountInterface {
+            .mast_forest()
+            .procedure_digests()
+            .collect::<Vec<Digest>>();
+
+        debug_assert!(rpo_falcon_procs.len() == 1);
+        let rpo_falcon_proc = rpo_falcon_procs[0];
+
+        procedures.iter().for_each(|proc_info| {
+            if proc_info.mast_root() == &rpo_falcon_proc {
+                component_interface_set
+                    .insert(AccountComponentInterface::RpoFalcon512(proc_info.storage_offset()));
+            }
+        });
+
+        let mut custom_interface_procs_map: BTreeMap<u8, Vec<AccountProcedureInfo>> =
+            BTreeMap::new();
+        procedures.iter().for_each(|proc_digest| {
+            if !basic_wallet_library()
+                .mast_forest()
+                .procedure_digests()
+                .any(|wallet_proc_digest| &wallet_proc_digest == proc_digest.mast_root())
+                && !basic_fungible_faucet_library()
+                    .mast_forest()
+                    .procedure_digests()
+                    .any(|faucet_proc_digest| &faucet_proc_digest == proc_digest.mast_root())
+                && (&rpo_falcon_proc != proc_digest.mast_root())
+            {
+                match custom_interface_procs_map.get_mut(&proc_digest.storage_offset()) {
+                    Some(proc_vec) => proc_vec.push(proc_digest.clone()),
+                    None => {
+                        custom_interface_procs_map
+                            .insert(proc_digest.storage_offset(), vec![proc_digest.clone()]);
+                    },
+                }
+            }
+        });
+
+        if !custom_interface_procs_map.is_empty() {
+            for proc_vec in custom_interface_procs_map.into_values() {
+                component_interface_set.insert(AccountComponentInterface::Custom(proc_vec));
+            }
+        }
+
+        component_interface_set
+    }
+
+    // probably it would be convenient to also have `from_components` constructor
+}
+
+// ACCOUNT COMPONENT INTERFACE
+// ================================================================================================
+
+/// Describes whether a note is compatible with a specific account.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum NoteAccountCompatibility {
+    /// A note is fully compatible with an account.
+    ///
+    /// Being fully compatible with an account interface means that account has all necessary
+    /// procedures to consume the note and the correctness of the note script is guaranteed,
+    /// however there is still a possibility that account may be in a state where it won't be
+    /// able to consume the note.
+    Yes,
+    /// A note is incompatible with an account.
+    ///
+    /// The account interface does not have procedures for being able to execute at least one of
+    /// the program execution branches.
+    No,
+    /// The account has all necessary procedures to consume the note, but the correctness of the
+    /// note script is not guaranteed.
```

Comment:
I think instead of correctness I would write it something like
> "The account has all necessary procedures of one execution branch of the note script. This means the note may be able to be consumed by the account if that branch is executed."